src/components/calcite-input/calcite-input.tsx

```diff
@@ -199,19 +199,21 @@ export class CalciteInput implements LabelableComponent, FormComponent {
 
   @Watch("value")
   valueWatcher(newValue: string): void {
-    if (newValue == null) {
-      this.value = "";
-      return;
-    }
-
-    if (
-      this.type === "number" &&
-      this.localizedValue !== localizeNumberString(newValue, this.locale)
-    ) {
-      this.setLocalizedValue(newValue);
-    } else if (this.childEl && this.childEl.value !== newValue) {
-      this.childEl.value = newValue;
+    if (!this.internalValueChange) {
+      this.setValue({
+        origin: "external",
+        value:
+          newValue == null
```

Comment:
Not crucial for this PR, but `setValue` already has some similar checks. Could these be moved there too? Another potential benefit would be that the invalid number warning check could be moved and simplified there since it already checks for valid numbers.